### ISSUE TITLE
feat: gradient accumulation canary (PMAT-459, PMAT-428)

### ISSUE THREAD

**Issue by noahgift** [paiml/qwen-train-canary#17]
## Problem

F-EXEC-02: Full fine-tune OOMs on yoga 8GB at batch=4. Gradient accumulation
(batch=1, accum=4) enables effective batch=4 with ~60% less VRAM.

## Implementation

- Added `--gradient-accumulation-steps N` to pytorch canary
- Loss scaled by 1/N per micro-batch, optimizer.step() every N steps
- Canary outputs `gradient_accumulation_steps` in config

## Contract

`gradient-accumulation-canary-v1.yaml`:
- 3 invariants (effective batch, gradient equivalence, VRAM proportional)
- 2 hypotheses (fits yoga 8GB, throughput penalty < 20%)
- 3 falsification tests

## Usage

```bash
# Gradient accumulation (fits yoga 8GB)
python canaries/pytorch/train.py --batch-size 1 --gradient-accumulation-steps 4

# Baseline comparison (gx10 only)
python canaries/pytorch/train.py --batch-size 4
```

Refs PMAT-459, PMAT-428
